File: Difficulty: Easy/Check Prime/check-prime.py
#{ 
 # Driver Code Starts
#Initial Template for Python 3

# } Driver Code Ends

# User function Template for python3

def prime(n):
    # Check if n is less than 2, as numbers less than 2 are not prime
    if n < 2:
        return False

    # Loop through all numbers from 2 to n-1
    for i in range(2, n):
        # If n is divisible by any of these numbers, it's not prime
        if n % i == 0:
            return False
    
    # If no divisor is found, n is prime
    return True



#{ 
 # Driver Code Starts.

if __name__ == '__main__':
    t=int(input())
    for _ in range(t):
        n=int(input())
        
        ans = prime(n)
        print(ans)
        print("~")
# } Driver Code Ends

File: Difficulty: Easy/Check Prime/check-prime.java
//{ Driver Code Starts

// } Driver Code Ends

import java.util.*;

public class Main {
    public static void main(String args[]) {
        // Create a scanner object to read input
        Scanner sc = new Scanner(System.in);
        
        // Read the integer value n
        int n = sc.nextInt();
        
        // Initialize a boolean variable to track if n is prime
        boolean isPrime = true;
        
        // Special case for n = 1, which is not prime
        if (n == 1) {
            isPrime = false;
        }
        // Special cases for n = 2 and n = 3, which are prime
        else if (n == 2 || n == 3) {
            isPrime = true;
        }
        // Check if n is divisible by 2 or 3, in which case it's not prime
        else if (n % 2 == 0 || n % 3 == 0) {
            isPrime = false;
        } else {
            // Check for divisibility starting from 5, incrementing by 6 (to check i and i+2)
            for (int i = 5; i * i <= n; i = i + 6) {
                // If n is divisible by i or i + 2, it's not prime
                if (n % i == 0 || n % (i + 2) == 0) {
                    isPrime = false;
                }
            }
        }
        
        // Output "True" if n is prime, otherwise output "False"
        if (isPrime) {
            System.out.println("True");
        } else {
            System.out.println("False");
        }
    }
}



//{ Driver Code Starts.
// } Driver Code Ends